spec/models/miq_server/configuration_management_spec.rb

```diff
@@ -83,5 +83,24 @@
         expect(message.method_name).to eq("reload_settings")
       end
     end
+
+    describe "#config_activated" do
+      let(:zone) { FactoryGirl.create(:zone, :name => "My Zone") }
+      let(:zone_other_region) do
+        other_region_id = ApplicationRecord.id_in_region(1, MiqRegion.my_region_number + 1)
+        FactoryGirl.create(:zone, :id => other_region_id).tap do |z|
+          z.update_column(:name, "My Zone") # Bypass validation for test purposes
+        end
+      end
+
+      it "saves settings with the zone in the correct region" do
+        miq_server = EvmSpecHelper.local_miq_server(:zone => zone)
+        zone_other_region
+
+        miq_server.config_activated(OpenStruct.new(:zone => "My Zone"))
+
+        expect(miq_server.reload.zone).to eq(zone)
```

Comment:
I don't know, I'm ok with writing the test for the current code.  The fact that database would return the "unexpected" one sporadically in the old code is unimportant if the current code always passes.  Thoughts?